### PR TITLE
Hover to expand mod customise panel

### PR DESCRIPTION
**Pull request by Cai1Hsu**
I didn't see anyone asking for this feature, but it feels fluent for me to select and adjust mods quickly.

The only place where I feel bad is that you have to click the header twice(collapse and expand the panel) to keep the panel expanded. I like that when hovering the header, you can click it to "pin" the panel so that it will not collapse when the cursor is left, and click it again to collapse it, as shown in the video at about 20s. but I didn't apply the [patch](https://gist.github.com/Cai1Hsu/c24a1da8b34d0be2254317b828050b6e) in the commits because I don't know if people like it.

## video demo

https://github.com/user-attachments/assets/e252c136-4d23-408f-8081-c39c151bc903

### CODE REVIEW

**Comment by shiumano** [comment]
Faster customization would be nice, but it seems a bit busy due to large panels opening and overlapping shadows. I would like a moment of grace.
Also, with touch input, where hover and click occur simultaneously, the concern about having to click twice on the header (collapsing and expanding the panel) is noticeable.

https://github.com/user-attachments/assets/02e7d2c0-1fb0-484f-8873-b2e3bf5a7990

**Comment by peppy** [comment]
I like this, but mobile obviously needs a fix.

**Comment by Cai1Hsu** [comment]
> Also, with touch input, where hover and click occur simultaneously, the concern about having to click twice on the header (collapsing and expanding the panel) is noticeable.

Thanks, I'm going to push a fix.

> but it seems a bit busy due to large panels opening and overlapping shadows. I would like a moment of grace.

could you make it more explicit? any suggestion for improvement?

File: osu.Game/Overlays/Mods/ModCustomisationHeader.cs [hunk 1]
```diff
@@ -28,6 +29,8 @@ public partial class ModCustomisationHeader : OsuHoverContainer
 
         public readonly BindableBool Expanded = new BindableBool();
 
+        protected new ModCustomisationPanel? Parent => (ModCustomisationPanel?)base.Parent;
```
Comment (bdach) [inline osu.Game/Overlays/Mods/ModCustomisationHeader.cs:32]: This is dodgy. Pass the parent via the constructor instead.

File: osu.Game/Overlays/Mods/ModCustomisationPanel.cs [hunk 2]
```diff
@@ -203,9 +224,20 @@ protected override void Update()
         private partial class FocusGrabbingContainer : InputBlockingContainer
         {
             public IBindable<bool> Expanded { get; } = new BindableBool();
+            public IBindable<bool> ExpandedByHovering { get; } = new BindableBool();
 
             public override bool RequestsFocus => Expanded.Value;
             public override bool AcceptsFocus => Expanded.Value;
+
+            public new ModCustomisationPanel? Parent => (ModCustomisationPanel?)base.Parent;
```
Comment (bdach) [inline osu.Game/Overlays/Mods/ModCustomisationPanel.cs:232]: As above

**Review by bdach — CHANGES_REQUESTED** [review]

**Review by peppy — DISMISSED** [review]

**Review by bdach — APPROVED** [review]

**Pull request merged**